### PR TITLE
SG-17257 Fix for deprecation of user.name in Rest API

### PR DESCRIPTION
**Pull request by eshokrgozar**
Fixes a couple of issues related to the deprecation of the user name in the Jira rest api

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -2598,7 +2598,7 @@ def search_assignable_users_for_issues(self,
         :rtype: ResultList
         """
         params = {
-            'username': username}
+            'query': username}
```
Comment (jfboismenu) [inline jira/client.py]: Does this work with super old versions of self-hosted JIRA?

Comment (eshokrgozar) [reply jira/client.py]: It wasn't causing an error but it was actually returning all the users and not really doing a match. Pushed a new fix.

**Review by jfboismenu — APPROVED** [review]
🌈

**Review by jfboismenu — APPROVED** [review]
Looks good to me!

**Comment by carlos-villavicencio-adsk** [comment]
Changes are already supported on upstream repo [3.5.0](https://github.com/pycontribs/jira/releases/tag/3.5.0):

- https://github.com/pycontribs/jira/commit/2c718f1130de0d1c274a778e9f053c01e8bbe50a
- https://github.com/pycontribs/jira/commit/502e554826216fc379cbef758c43ea3f5f166336
- https://github.com/pycontribs/jira/commit/e68bf79ec76c927f15e57e3fddd025e8028fe6fc

**Pull request merged**